python/cudf/cudf/core/series.py

```diff
@@ -1194,17 +1194,7 @@ def where(self, cond, other=None, axis=None):
         to_replace = self._column.apply_boolean_mask(~cond & self.notna())
         if is_scalar(other):
             all_nan = other is None
-            if all_nan:
-                new_value = [other] * len(to_replace)
-            else:
-                # pre-determining the dtype to match the pandas's output
-                typ = to_replace.dtype
-                if np.dtype(type(other)).kind in "f" and typ.kind in "i":
-                    typ = np.int64 if other == int(other) else np.float64
-
-                new_value = utils.scalar_broadcast_to(
-                    other, (len(to_replace),), np.dtype(typ)
-                )
+            new_value = [other] * len(to_replace)
```

Comment:
Not sure how large `to_replace` or `replacement` are expected to be in real world cases. But I am opposed to using this approach since `[other] * len(to_replace)` is much slower than `utils.scalar_broadcast` for larger sizes.

The reason I did this is so that the object remains a python list/scalar at the `find_and_replace_call` and we can identify the min viable dtype. If the columns are strongly typed objects like np arrays or device arrays we do nothing.
To retain the `scalar_broadcast` method would mean moving that logic down to the `replace` method in `series.py` which I am hesitant to implement without confirmation since it involves a special branch for handling numeric types. Despite this generic function working with all dtypes.

Reply:
Looks like this can easily be avoided just by having libcudf support for scalars

Reply:
Any recommendation on how to generally proceed. 
- One approach is having all the dtype identification/promotion logic within `find_and_replace` but then we need to pass a python list/scalar from `series.replace` to `numeric.find_and_replace`.
- Second option within `series.replace` check if inputs are numeric and perform the dtype casting operations there. (not as clean but more performant)
- Third option that I seem to understand from your suggestion. libcudf handles scalars for `replace`?

Reply:
I'm not sure I understand the details relating to typecasting (I admit I haven't reviewed this deeply and followed the discussion above). Why should a list be handled any differently than a `Column`?

I was only referring to the issue relating to scalar broadcasting, which can be avoided completely by adding libcudf support for scalars in `find_and_replace_all`, which it does not currently have.